doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +554,140 @@ include the chapel runtime and standard modules for use in a non-Chapel program
+other settings in the future, see :ref:`Other Settings` in the :ref:`Multilocale
+Caveats` section for more information.
+
+To compile a multilocale library, `ZeroMQ <http://zeromq.org/>`_ must be
+installed.
+
+If ZeroMQ is not installed in a way that enables your C compiler to find it
+easily, the environment variable ``CHPL_ZMQ_HOME`` can be set.  This environment
+variable should be set to a directory containing both an ``include`` directory
+which contains ``zmq.h`` and a ``lib`` directory which contains ``libzmq.*``.
+For example, for a directory structure:
+
+.. code-block:: text
+
+   |-- .local/
+   |    |-- include/
+   |    |    |-- zmq.h
+   |    |-- lib/
+   |    |    |-- libzmq.a
+   |    |    |-- libzmq.so
+
+``CHPL_ZMQ_HOME`` would be set to ``/absolute/path/to/.local/``.
+
+Initializing Your Multilocale Library
+-------------------------------------
+
+From the user's perspective, a strategy similar to that of single locale
+libraries is used when initializing a multilocale library.  However, like
+transitioning between a single locale executable and a multilocale one, users
+must now specify the number of locales required for the multilocale library.
+
+In C
+~~~~
+
+Users must still call ``chpl_library_init()`` before utilizing the exported
+Chapel functions.  However, the ``char* argv[]`` must now include two additional
+entries: the numlocales flag and its intended value.
+
+This can be accomplished either by explicitly adding the arguments in the C
+client program itself, or by passing them as arguments to the executable.
+
+The following example programs use the ``foo`` library.  This first example
+demonstrates explicitly adding the arguments in the program:
+
+.. code-block:: C
+
+   #include "foo.h"
+
+   int main(int argc, char* argv[]) {
+     int argChapelC = 3;
+     char* argChapelV[3] = {argv[0], "-nl", "2"};
+     // Initialize the Chapel runtime and standard modules
+     chpl_library_init(argChapelC, argChapelV);
+
+     baz(7); // Call into a library function
+
+     chpl_library_finalize();
+
+     return 0;
+   }
+
+Alternatively, the original single locale client from `Initializing Your
+Library In C`_ can be used with the additional two arguments to the executable:
+
+.. code-block:: bash
+
+   ./a.out -nl 2
+
+Users also still need to call the generated module initialization function for
+multilocale libraries, as mentioned in that section.
+
+In Python
+~~~~~~~~~
+
+Users must still call ``chpl_setup()`` before utilizing the exported Chapel
+functions.  However, it requires a ``numLocales`` argument when the library
+has been compiled for multilocale settings.
+
+Makefile-less Compilation
+-------------------------
+
+When compiling a C program using a multilocale Chapel library without a
+makefile, some additional steps are needed beyond those required for using a
+:ref:`single locale Chapel library <Makefileless Compilation In Single Locale>`.
```

Comment:
Should you say that using a Makefile might be worth the time, since it simplifies this use case significantly?

Reply:
How does this update look?  (I still have a couple of other outstanding TODOs)